src/intree.js

```diff
@@ -54,54 +56,57 @@ function completeInTreeConfig(config, payload) {
+    let errors = validator(config, schema);
+    if (errors) {
+      throw new Error(errors);
+    }
+    debug(`intree config for ${payload.organization}/${payload.repository} matches valid schema.`);
 
-  // We need to toss out the config version number; it's the only
-  // field that's not also in graph/task definitions
-  let version = config.version;
-  delete config.version;
+    // We need to toss out the config version number; it's the only
+    // field that's not also in graph/task definitions
+    let version = config.version;
+    delete config.version;
 
-  // Perform parameter substitutions. This happens after verification
-  // because templating may change with schema version, and parameter
-  // functions are used as default values for some fields.
-  config = jparam(config, _.merge(payload.details, {
-    $fromNow: (text) => tc.fromNowJSON(text),
-    organization: payload.organization,
-    repository: payload.repository,
-    'taskcluster.docker.provisionerId': 'aws-provisioner-v1',
-    'taskcluster.docker.workerType': 'github-worker',
-  }));
+    // Perform parameter substitutions. This happens after verification
+    // because templating may change with schema version, and parameter
+    // functions are used as default values for some fields.
+    config = jparam(config, _.merge(payload.details, {
+      $fromNow: (text) => tc.fromNowJSON(text),
+      organization: payload.organization,
+      repository: payload.repository,
+      'taskcluster.docker.provisionerId': cfg.intree.provisionerId,
+      'taskcluster.docker.workerType': cfg.intree.workerType,
+    }));
 
-  // Compile individual tasks, filtering any that are not intended
-  // for the current github event type
-  config.tasks = config.tasks.map((task) => {
-    return {
-      taskId: slugid.nice(),
-      task: task,
-    };
-  }).filter((task) => {
-    // Filter out tasks that aren't associated with the current event
-    // being handled
-    let events = task.task.extra.github.events;
-    let branches = task.task.extra.github.branches;
-    return _.some(events, ev => RegExp(ev).test(payload.details['event.type'])) && (!branches || branches &&
-      _.includes(branches, payload.details['event.base.repo.branch']));
-  });
-  return completeInTreeConfig(config, payload);
+    let taskGroupId = slugid.nice();
+    // Compile individual tasks, filtering any that are not intended
+    // for the current github event type. Append taskGroupId while
+    // we're at it.
+    config.tasks = config.tasks.map((task) => {
+      return {
+        taskId: slugid.nice(),
+        task: _.extend(task, {taskGroupId}),
+      };
+    }).filter((task) => {
+      // Filter out tasks that aren't associated with the current event
+      // being handled
+      let events = task.task.extra.github.events;
+      let branches = task.task.extra.github.branches;
+      return _.some(events, ev => RegExp(ev).test(payload.details['event.type'])) && (!branches || branches &&
```

Comment:
For example: `RegExp(ev)` where `ev` is untrusted is actually an attack vector... It's relatively easy to construct regular expressions that takes forever to evaluate, hence, crashing the server -- just some fun to think about :)

Note: not saying that it should be done in this PR. Just saying lots of low hanging fruit in tc-gh.